### PR TITLE
Remove deprecated imperative constants, options, and methods - Imperative

### PR DESCRIPTION
**Pull request by ATorrise**
**What It Does**
Removes the following items marked as deprecated:
- [x] [flattenCommandTreeWithAliases](https://github.com/zowe/imperative/blob/master/packages/cmd/src/utils/CommandUtils.ts#L176) 
- [x] [requireKeytar](https://github.com/zowe/imperative/blob/master/packages/config/src/doc/IProfOpts.ts#L37)
- [x] [envVariablePrefix](https://github.com/zowe/imperative/blob/master/packages/imperative/src/Imperative.ts#L472)
- [x] [AbstractAuthHandler.getPromptParams](https://github.com/zowe/imperative/blob/master/packages/imperative/src/auth/handlers/AbstractAuthHandler.ts#L71)
- [x] [BaseAuthHandler.getPromptParams](https://github.com/zowe/imperative/blob/master/packages/imperative/src/auth/handlers/BaseAuthHandler.ts#L57)
- [x] [pluginHealthCheck](https://github.com/zowe/imperative/blob/master/packages/imperative/src/doc/IImperativeConfig.ts#L264)
- [x] [promptForInput and promptWithTimeout](https://github.com/zowe/imperative/blob/master/packages/utilities/src/CliUtils.ts)
- [x] Remove TODO comment regarding PMF_ROOT in the [PluginManagementFacility](https://github.com/zowe/imperative/blob/master/packages/imperative/src/plugins/PluginManagementFacility.ts#L228)

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

> [!NOTE]  
> 1. Changes were grouped into different commits to aid the review process.
> 2. This PR addresses [zowe/imperative PR 970](https://github.com/zowe/imperative/issues/970) but also internally [Rally US925239](https://rally1.rallydev.com/#/227079578160d/teamboard/list?detail=%2Fuserstory%2F730627401759&view=91f318be-db4f-4d2a-85e7-263bf2a96459). Rally has task to close out any related issues that are now irrelevant because of the deprecation. My method for determining relevant issues was to search issues on github with the names of the packages I've deprecated as a keyword. Any results I investigated. This method produced no issues to close out.

### CODE REVIEW

**Comment by ATorrise** [comment]
@zFernand0 should i update the different changelogs and say what's been removed since it was deprecated?

**Comment by t1m0thyj** [comment]
> @zFernand0 @gejohnston @t1m0thyj should i update the different changelogs and say what's been removed?

Yes please 🙂 You can see an example of how we did it when deprecating old stuff in v1 [here](https://github.com/zowe/zowe-cli/blob/master/packages/imperative/CHANGELOG.md#500-next202106041929):
image

**Comment by zFernand0** [comment]
Since the merging order caused this, I'm happy to help with merge conflicts 😋

**Review by zFernand0 — APPROVED** [review]
LGTM! :yum:

File: packages/imperative/src/imperative/src/doc/IImperativeConfig.ts [hunk 1]
```diff
@@ -233,31 +233,6 @@ export interface IImperativeConfig {
...
-     *      This property is deprecated. Plugins that want to perform a health check can
-     *      specify the pluginLifeCycle property to load a class from the plugin.
-     *      The plugin can implement the postInstall() function of that class to perform
-     *      a health check, or any other desired operation. Our intention is to remove
-     *      the pluginHealthCheck property in Zowe V3.
-     *
-     * @type {string}
-     * @memberof IImperativeConfig
-     */
-    pluginHealthCheck?: string;
```
[the remark this answers is not in the document]
Comment (ATorrise) [reply packages/imperative/src/imperative/src/doc/IImperativeConfig.ts:259]: @gejohnston Sounds good. In a zowe wide search for pluginHealthCheck I get results in zowe-cli-sample-plugin, zowe-cli-scs-plugin, and the api-layer. Will create PRs for each of those repos and link back to this PR

**Pull request merged**